src/main/java/org/opensearch/ad/rest/RestDeleteAnomalyDetectorAction.java

```diff
@@ -105,7 +106,7 @@ public List<ReplacedRouteHandler> replacedRouteHandlers() {
             );
     }
 
-    private Function<RestRequest, ExtensionRestResponse> handleRequest = (request) -> {
+    private Function<RestRequest, RestResponse> handleRequest = (request) -> {
```

Comment:
Trying to understand why are we changing the response type from `ExtensionRestResponse` to `RestResponse`?
If this is coming from NameRoute then we should change the return type of `prepareRequest` to `RestResponse` as well?

Reply:
NamedRoute is generally available to both plugins and extensions, and hence RestResponse instead of ExtensionRestResponse (which inherits RestReponse transitively). Which instance of `prepareRequest` are you referring to?

Reply:
On line 71

Reply:
prepareRequest should still return ExtensionRestResponse as it is one of the concrete implementations of RestResponse and will be interpreted as such when send the response back to user. However, the only use case I see for ExtensionRestResponse is for communication between OS and extension to mark that the content was consumed and a list of consumed content. 

@dbwiddis had some thoughts on whether we should keep ExtensionRestResponse or just use BytesRestReponse class